clippy_lints/src/operators/arithmetic_side_effects.rs

```diff
@@ -98,8 +98,15 @@ impl ArithmeticSideEffects {
     }
 
     /// If `expr` is not a literal integer like `1`, returns `None`.
+    ///
+    /// It is worth nothing that the returned number can or can not be negative.
     fn literal_integer(expr: &hir::Expr<'_>) -> Option<u128> {
-        if let hir::ExprKind::Lit(ref lit) = expr.kind && let ast::LitKind::Int(n, _) = lit.node {
+        let actual = if let hir::ExprKind::Unary(hir::UnOp::Neg, elem) = expr.kind {
+            elem
+        } else {
+            expr
+        };
```

Comment:
Tiny nit, which is probably not worth implementing, but I wanted to mention it. Rust allows multiple unary operators. A user could write `--1` which would return `None`. However, since this is only used to suppress the lint if the expression is definitely safe, I don't think it's important to handle this case :)

Reply:
Created an external function that "peels" unary operators,